firebase-perf/src/main/java/com/google/firebase/perf/session/FirebasePerformanceSessionSubscriber.kt

```diff
@@ -19,28 +19,16 @@ package com.google.firebase.perf.session
 import com.google.firebase.perf.session.gauges.GaugeManager
 import com.google.firebase.perf.v1.ApplicationProcessState
 import com.google.firebase.sessions.api.SessionSubscriber
-import java.util.UUID
 
 class FirebasePerformanceSessionSubscriber(override val isDataCollectionEnabled: Boolean) :
   SessionSubscriber {
 
   override val sessionSubscriberName: SessionSubscriber.Name = SessionSubscriber.Name.PERFORMANCE
 
   override fun onSessionChanged(sessionDetails: SessionSubscriber.SessionDetails) {
-    val currentPerfSession = SessionManager.getInstance().perfSession()
-
-    // A [PerfSession] was created before a session was started.
-    if (currentPerfSession.aqsSessionId() == null) {
-      currentPerfSession.setAQSId(sessionDetails)
-      GaugeManager.getInstance()
-        .logGaugeMetadata(currentPerfSession.aqsSessionId(), ApplicationProcessState.FOREGROUND)
-      return
-    }
-
-    val updatedSession = PerfSession.createWithId(UUID.randomUUID().toString())
-    updatedSession.setAQSId(sessionDetails)
+    val updatedSession = PerfSession.createWithId(sessionDetails.sessionId)
     SessionManager.getInstance().updatePerfSession(updatedSession)
     GaugeManager.getInstance()
```

Comment:
double check the logic here

Reply:
Defering to @mrober - but this seems to assume that an AQS will always be available w/ `DebugEnforcementCheck` asserting it.

While it seems feasible, I think this is a little pre-emptive.

Additionally none of the relevant tests (especially `PerfSessionTest`) doesn't include a test with `enforcement` = `false` - which adds more complexity to this change.

I'm biased towards keeping this behavior as is for now - and following it up with a separate PR.

Reply:
K revert the change here